www/src/pages/insights.jsx

```diff
@@ -171,7 +173,29 @@ const ComponentsUsage = () => (
   </div>
 );
 
-export default function InsightsPage() {
+export default function InsightsPage({ location }) {
+  const tabs = ['/insights', '/insights/?tab=projects', '/insights/?tab=components'];
+  const path = `${location.pathname}${location.search}`;
+
+  const handleOnSelect = (value) => {
+    if (value !== path) {
+      navigate(value);
+      switch (value) {
+        case tabs[0]:
+          window.analytics.track('Usage Insights (Summary)');
```

Comment:
When the SEGMENT_KEY env variable is not set, are you seeing `global.analytics` as undefined? I'm getting errors locally calling `.track()` on undefined when SEGMENT_KEY isn't set. Any ideas on how to get around this?

Reply:
Adding conditional plugin connection depending on the `SEGMENT_KEY `